src/Azure.Functions.Cli/Actions/AzureActions/PublishFunctionAppAction.cs

```diff
@@ -1040,51 +1023,55 @@ private async Task<string> UploadPackageToStorage(Stream package, string blobNam
 
                 const string containerName = "function-releases";
 
-                CloudBlobContainer blobContainer = null;
-
+                BlobContainerClient blobContainerClient = null;
                 try
                 {
-                    var storageConnection = appSettings["AzureWebJobsStorage"];
-                    var storageAccount = CloudStorageAccount.Parse(storageConnection);
-                    var blobClient = storageAccount.CreateCloudBlobClient();
-                    blobContainer = blobClient.GetContainerReference(containerName);
-                    await blobContainer.CreateIfNotExistsAsync();
+                    if (appSettings.TryGetValue(Constants.AzureWebJobsStorageAccountName, out var accountName))
+                    {
+                        var storageAccount = await ArmClient.FindStorageAccount(accountName);
+                        blobContainerClient = new BlobContainerClient(new UriBuilder(storageAccount.Properties.PrimaryEndpoints.Blob) { Path = containerName }.Uri,
+                            new Storage.StorageSharedKeyCredential(accountName, storageAccount.Key.Value));
```

Comment:
Remove the SotrageSharedKeyCredential and use this.Credential instead.
This allows disabling Shared Keys on the storage account altogether.